### PR TITLE
Courses model

### PR DESCRIPTION
**Pull request by benguarasci**
Courses model/serializer and initial route

### CODE REVIEW

File: scheduler/urls.py [hunk 1]
```diff
@@ -19,5 +19,6 @@
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('api/', include('users.urls')),
+    path('api/', include('courses.urls')),
```
Comment (Russell-Waterhouse) [inline scheduler/urls.py:22]: Looks like there's a merge conflict in this file

File: courses/models.py [hunk 2]
```diff
@@ -1,3 +1,45 @@
...
+        db_table = 'Courses'
+
+
+
+
+#Algorithm Specific Models
+
+
+
+# class CourseSection(models.Model):
```
Comment (Russell-Waterhouse) [inline courses/models.py:28]: Is this commented-out code needed for something later?

**Review by Russell-Waterhouse — APPROVED** [review]
Looks good to me, just address those 2 comments I left about merge conflicts and commented-out code and merge it in!

File: courses/models.py (hunk 2 above)
Comment (benguarasci) [reply courses/models.py:32]: Yeah its for the Algorithm stuff that Im working on this sprint

**Comment by rbassot** [comment]
Is the `section` field intended for this model? From the FE spec, it seems like we never discussed implementing this field (though if you look at this document, it seems slightly out of date):
image
image

File: courses/serializers.py [hunk 3]
```diff
@@ -1,3 +1,49 @@
...
+        except ValidationError:
+            raise serializers.ValidationError({"error": "invalid input"})
+        except Course.models.DoesNotExist:
+            raise serializers.ValidationError({"error": "this course object does not exist"})
+        return course
+
+   
+    def update(self, instance, validated_data):
+        """
+        Update and return an existing Preferences instance, given the validated data.
```
Comment (rbassot) [inline courses/serializers.py:36]: Could clean up some of these docstrings to remove any mention of `Preferences`

**Review by rbassot — APPROVED** [review]

**Pull request merged**